wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/WorkspaceRuntimeImpl.java

```diff
@@ -94,6 +95,13 @@ public void setMachines(List<MachineImpl> machines) {
         this.machines = machines;
     }
 
+    public synchronized void addMachine(MachineImpl machine) throws ConflictException {
```

Comment:
1. This synchronization is ugly as long as you have another methods related to `this.machines` not synchornized, but even if you have those synchronizations it is still ugly. Remove synchronized from the method.
2. `ConflictException` ? did you see any model implementation which throws checked exceptions? i guess no.
Please remove `throws ConflictException`. I don't think that validations should be performed on this object level, as we have certain places for it(btw validation is also ugly as it may be performed in more lightweight way, and is it weak as it doesn't check all the things(e.g. that machine is not dev one)).

So basically you can add the `addMachine` method to the `WorkspaceRuntimeImpl` but all what this method should do is to add the machine to the list of the machines.